torch/_subclasses/fake_tensor.py

```diff
@@ -439,6 +439,14 @@ def resize_as_(fake_mode, func, *args, **kwargs):
         return func(*args, **kwargs)
 
 
+# Don't default to default device handling since range check is only applied to
+# scalar and doing the check as side effect.
+@register_op_impl(aten.sym_constrain_range.default)
```

Comment:
I think this is unnecessary. What happens if you remove this ?

Reply:
If I removed it, it will crash at https://github.com/pytorch/pytorch/blob/e82616d9001c6cc7a533b92eaab0b41eba5533e6/torch/_subclasses/fake_tensor.py#L1412 and complains "Could not find common device for aten.sym_constrain_range.default".

My gut feeling is that for our use case this function shouldn't run at all because `sym_constrain_range` returns `None` so it doesn't make sense to infer device for it?

Reply:
Im surprised you get this error because there is no Tensor output, right, so we shouldn't be calling the wrap function.

Reply:
@eellison: So this change looks OK then?

Reply:
@eellison: Awesome! It works now. Thanks for the help!